src/utiltest.cpp

```diff
@@ -71,6 +71,34 @@ CWalletTx GetValidSproutReceive(ZCJoinSplit& params,
                                 joinSplitPrivKey
                                ) == 0);
 
+    return mtx;
+}
+
+CWalletTx GetValidSproutReceive(ZCJoinSplit& params,
+                                const libzcash::SproutSpendingKey& sk,
+                                CAmount value,
+                                bool randomInputs,
+                                int32_t version /* = 2 */)
+{
+    CMutableTransaction mtx = GetValidSproutReceiveTransaction(
+        params, sk, value, randomInputs, version
+    );
+    CTransaction tx {mtx};
+    CWalletTx wtx {NULL, tx};
+    return wtx;
+}
+
+CWalletTx GetInvalidSproutReceive(ZCJoinSplit& params,
```

Comment:
This is a very specific kind of invalid, so this needs either renaming (to reflect the fact that it refers to invalid commitments), or refactoring (to use e.g. a lambda to make the transaction invalid that can then be test-specific, or just have this inlined in the test).